src/system/SystemPacketBuffer.cpp

```diff
@@ -355,6 +357,29 @@ PacketBuffer * PacketBuffer::Consume(uint16_t aConsumeLength)
     return lPacket;
 }
 
+CHIP_ERROR PacketBuffer::Read(uint8_t * aDestination, size_t aLength)
+{
+    PacketBuffer * lPacket = this;
+
+    if (aLength > TotalLength())
+    {
+        return CHIP_ERROR_BUFFER_TOO_SMALL;
+    }
+    while (lPacket != nullptr && aLength > 0)
+    {
+        size_t lLength = lPacket->DataLength();
+        if (aLength < lLength)
+        {
+            lLength = aLength;
+        }
+        memcpy(aDestination, lPacket->payload, lLength);
```

Comment:
Why not `lPacket->Start()`?  I would vaguely prefer we use the public API as long as it does what we want and is zero-cost, which should be true here.